### DISCUSSION TITLE
FerrisChat Discussion!

### DISCUSSION THREAD

**Discussion by hydrostaticcog** [FerrisChat/FerrisChat#16] [General]
Use this thread to talk about FerrisChat with @tazz4843.

**Comment by asze17** [comment]
How do you send an event to client with ws in other files like in send_message.rs?

**Reply by tazz4843** [reply]
I have no idea yet. I'll figure that out this weekend.

**Reply by tazz4843** [reply]
I can now give you an answer: serialize the event with `rmp-serde`, send it via Redis pubsub, and have code in the WS handler that does final checks and sends the event to the client.

**Comment by hydrostaticcog** [comment]
@tazz4843 im getting a build error. `hydro@node1:~/Server$ cargo build --release
   Compiling ferrischat_auth v0.1.0 (/home/hydro/Server/ferrischat_auth)
error[E0437]: type Config is not a member of trait FromRequest
  --> ferrischat_auth/src/auth_struct.rs:12:5
   |
12 |     type Config = ();
   |     ^^^^^^^^^^^^^^^^^ not a member of trait FromRequest

For more information about this error, try rustc --explain E0437.
error: could not compile ferrischat_auth due to previous error
hydro@node1:~/Server$ 
`

**Reply by hydrostaticcog** [reply]
Fixed that, new error:

hydro@node1:~/Server$ cargo build --release
   Compiling ferrischat_ws v0.1.0 (/home/hydro/Server/ferrischat_ws)
error[E0599]: no variant named `Resume` found for enum `WsInboundEvent`
   --> ferrischat_ws/src/lib.rs:199:71
    |
199 |                     WsInboundEvent::Identify { .. } | WsInboundEvent::Resume { .. } => {}
    |                                                                       ^^^^^^ variant not found in `WsInboundEvent`

error[E0599]: no variant named `Resume` found for enum `WsInboundEvent`
   --> ferrischat_ws/src/lib.rs:310:33
    |
310 |                 WsInboundEvent::Resume {
    |                                 ^^^^^^ variant not found in `WsInboundEvent`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `ferrischat_ws` due to 2 previous errors

**Reply by hydrostaticcog** [reply]
im fairly certain you actually removed that variant in a commit to the common repo.

**Reply by tazz4843** [reply]
hmmmm, i swear i removed that

anyways commit 1afbeff1adc7a57c125fff3683ede63603478ce3 removes it

**Reply by hydrostaticcog** [reply]
thanks, its building

**Reply by hydrostaticcog** [reply]
Hi, now we are getting a 500

heres the error:
POST https://api.ferris.chat/v0/channels/1002544655807437714533333860352/messages Returned 500 with {"reason":"Failed to serialize message to msgpack format: u128 is not supported"}

**Reply by tazz4843** [reply]
49ba0291733e041be2f4944c6bd15b7b4effa749 fixes

i take from this we cannot use msgpack whatsoever

**Reply by hydrostaticcog** [reply]
new error time:

500 on POST https://api.ferris.chat/v0/channels/973714649547311070802310332416/messages
{
    "reason": "Redis returned an error: Response was of incompatible type: \"Response type not string compatible.\" (response was int(0))"
}

**Reply by hydrostaticcog** [reply]
you really need to test this stuff before you push

**Reply by hydrostaticcog** [reply]
PTA is up to date

**Reply by tazz4843** [reply]
my bad it was 6e549796364c4b3d741dd0beb3097989225e7e72

**Reply by hydrostaticcog** [reply]
thanks, not getting a 500 anymore, now to see if we receive an event on WS correctly

**Reply by asze17** [reply]
Send message works but didn't receive event on WS.

**Reply by tazz4843** [reply]
send guild ID and start spamming send message while i keep a eye on redis

**Reply by asze17** [reply]
guild is is 1002542742713912450205276438528

**Reply by asze17** [reply]
And I have started spamming send message.

**Reply by tazz4843** [reply]
alright, messages are getting sent to redis: this points to a issue with the WS end

issue for tomorrow, through, as it's 2045 and i'm watching the federal election coverage

**Reply by asze17** [reply]
So did you fixed it? or

**Reply by hydrostaticcog** [reply]
not yet

**Comment by hydrostaticcog** [comment]
Rust 2021 Edition has arrived! Here are the repos that are being updated, and the PRs associated:

- [X] FerrisChat/Common#23
- [x] FerrisChat/SnowflakeGenerator#1
- [x] FerrisChat/Server#32
